Issue143

jnicholCU opened the pull request:
Newsletter page type addition. Does not have functionality to make the inlined css newsletter with php yet.
Has related pull requests in the paragraphs and page_types repos


Linked to #143

jnicholCU commented:
Related Pull Requests:

https://github.com/CuBoulder/ucb_custom_page_types/pull/26
https://github.com/CuBoulder/ucb_custom_paragraphs/pull/23

patrickbrown-io commented on templates/content/node--ucb-newsletter.html.twig:
```diff
@@ -0,0 +1,68 @@
+	<div class="container ucb-newsletter">
+		{# Newsletter Title #}
+		<h1 class="ucb-newsletter-title" itemprop="name">{{ content.field_newsletter_type }}</h1>
+
```
Needs the following : Date, View on Website link 
#143 see render order

jnicholCU replied:
Good catch! Definitely needs the date.
The "View on Website" link will be part of the rendering for the email version of the page.

jnicholCU replied:
Oh actually, just check the website render on the webexpress instructions.
Doesn't look like the date gets on that page either. Just for the emailed version.

https://websupport.colorado.edu/article/261-newsletter

patrickbrown-io commented on templates/content/node--ucb-newsletter.html.twig:
```diff
@@ -0,0 +1,68 @@
+				{% endif %}
+			</div>
+		</div>
+	</div>
```
Needs a footer that is unique to this newsletter type

patrickbrown-io reviewed (commented):
Functionality from admin side works following Kevin's specifications outlined on #143 

Needs:
- Date and Link to render after newsletter name
- Footer

patrickbrown-io commented:
One last one - the teaser thumbnail image is on the right on current teaser content, Kevin's mockup shows teaser-style content with image on left

jnicholCU commented:
> One last one - the teaser thumbnail image is on the right on current teaser content, Kevin's mockup shows teaser-style content with image on left

I just pinged Kevin about this one. His mockup seems different from what the webexpress site and options state in the creation instructions so Im going to get some more clarity 

The creation instructions state 
> Display: choose how articles in this section will display.
> Teaser: a 1/5 display with an article summary and a small thumbnail image aligned to the right.

And on the example addons page it has them displayed that way as well
https://www.colorado.edu/demos/web-express/newsletter/add-bundles/newsletter-example-one

Need to see if the teasers switch for the email view.

jnicholCU commented:
I also just noticed that the Promos section for the Newsletter just don't exist in this build for some reason? Looks like all the files are missing so I must have failed to commit those up.

patrickbrown-io commented:
I can test again once the promos are in

patrickbrown-io reviewed (commented):
Awesome -- promos work on fresh build and appear in the correct order in the newsletter, as specified! I agree with you that it looks a bit cluttered depending on how much content is put in the sections, so that extra css would help give everything a little breathing room.

 #143 
"Each newsletter has 2 promo areas which include an image and a link." =>  we might need a promo link / button instead of promo text? Or in addition to the promo text? I'm not sure which is the best course, no example of promos on [web express demos](https://www.colorado.edu/demos/web-express/newsletter/add-bundles/newsletter-example-one)

jnicholCU commented:
Wow yeah I don't know how I managed to build that wrong twice with text instead of a link. I believe the link will go on the promo so it's like an ad rather than a button. I'll get that changed up now so it's good for testing by the time you guys are up.

jnicholCU commented:
Updated with the promo links added in. Should be working fine.

patrickbrown-io reviewed (commented):
Promo images are clickable and in the correct place in the newsletter. CSS addition looks great. One thing, the teaser-style sections specifically on mobile screen size, might need to stack on mobile - it just gets VERY small fixed in a row. Other than that, looks great!

jnicholCU commented:
Added the change for the teaser layout so that those stack better on mobile
changed from `"col-5" & "col-1"` to `"col-lg-5 col-8" & "col-lg-1 col-4"`

patrickbrown-io reviewed (approved):
Nice! The mobile responsive teasers look great, approve!

jacob-korf commented:
Works great! Will be merged in
